srml/democracy/src/lib.rs

```diff
@@ -14,7 +14,143 @@
+//!  for mature referenda and tally their votes, then remove them from the Table of Referenda. If a referendum's
+//!  vote tally meets its vote threshold, then it will be passed and executed. Last, increment `NextTally`
+//!  to determine the next public referendum index to tally.
+//!
+//! ### Goals
+//!
+//! The Democracy module in Substrate is designed to make the following possible:
+//!
+//! - Create and sponsor public proposals.
+//! - Elevate public proposals to the Table of Referenda.
+//! - Start and cancel public referenda.
+//! - Vote on public referenda.
+//! - Delegate proxy voting rights of public referenda.
+//! - Tally votes of public referenda.
+//! - Pass and execute maturing public referenda.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! - `propose` - Propose a sensitive action to be taken.
+//! - `second` - Second (sponsor) a proposal.
+//! - `vote` - Vote in a referendum.
+//! - `proxy_vote` - Vote in a referendum on behalf of a stash.
+//! - `start_referendum` - Start a referendum.
+//! - `cancel_referendum` - Remove a referendum.
+//! - `cancel_queued` - Cancel a proposal queued for enactment.
+//! - `on_finalize` - Called when a block is finalized.
+//! - `set_proxy` - Specify a proxy. Called by the stash.
+//! - `resign_proxy` - Clear the proxy. Called by the proxy.
+//! - `remove_proxy` - Clear the proxy. Called by the stash.
+//! - `delegate` - Delegate vote.
+//! - `undelegate` - Undelegate vote.
+//!
+//! ### Public Functions
+//!
+//! - `locked_for` - Get the balance locked in support of a proposal.
+//! - `is_active_referendum` - Return true if given index corresponds to an on-going referendum.
+//! - `active_referendums` - Get all referenda that are currently active and their corresponding info.
+//!  Equivalent to the Table of Referenda.
+//! - `maturing_referendums_at` - Get all referenda ready for tally at block `n`.
+//! - `tally` - Tally the votes for the current proposal.
+//! - `force_proxy` - Forcibly insert a proxy voter for a stash account.
+//! - `internal_start_referendum` - Start a referendum. Can be called directly by the council.
+//! - `internal_cancel_referendum` - Remove a referendum. Can be called directly by the council.
+//!
+//! ## Usage
+//!
+//! ### Prerequisites
+//!
+//! Import the Democracy module and types and derive your runtime's configuration traits
+//! from the Democracy module trait.
+//!
+//! ### Example from the SRML
+//!
+//! The [Council module](../srml_council/index.html) uses the Democracy module for voting.
+//!
+//! ```
+//! use srml_support::{decl_module, dispatch::Result};
+//! # use srml_democracy as democracy;
+//! # use system::ensure_signed;
+//! # type VoteIndex = u64;
+//! pub trait Trait: democracy::Trait { }
+//!
+//! decl_module! {
+//! 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+//!
+//! 		fn proxy_set_approvals(origin, votes: Vec<bool>, index: VoteIndex) -> Result {
```

Comment:
nitpick: is this function indented too much?

Reply:
I think there are spacing issues in GitHub which do not appear when you open it in Visual Studio Code.